### ISSUE TITLE
[Feature Request] Rotating Proxy

### ISSUE THREAD

**Issue by BaseMax** [telegramdesktop/tdesktop#25145]
### Is your feature request related to a problem?

Hi there,
Based on removing Iran from USA internet sanctions, please enable telegram rotating proxy for Iranian.

### Describe the solution you'd like

Due to the filtering of WhatsApp and Instagram, a number of Iranian users returned to Telegram. If a rotating proxy is added, our problem will be solved. Because **MTProto** proxies are quickly filtered. And many users use unofficial telegrams, their security is low.

### Describe alternatives you've considered

Manually switch between proxies

**Comment by Aokromes** [comment]
https://github.com/kitabisa/mubeng

**Comment by BaseMax** [comment]
Thanks for sharing, There is any plan for internal support? It will be good if you can mention it on the board. @Aokromes

**Comment by pouriaksrvi** [comment]
> https://github.com/kitabisa/mubeng

The feature requested in this issue is different than what requested in my issue.
I asked for proxy rotation.
This user is asked for proxy rotation using AWS which you can understand from the explanation: "Based on removing Iran from USA internet sanctions" 

Please open my issue again.

Also the app you suggested doesn't support MTPROTO which is the main proxy protocol for Telegram because of its sponsored channel feature which the app obviously can never support it.

**Comment by BaseMax** [comment]
> This user is asked for proxy rotation using AWS which you can understand from the explanation

No, I have not talked about using AWS or any other service.

I suggested that Iranian users due to Telegram is banned internally need this feature and I think  Rotating Proxy is a good feature that can be useful.

Best,
M.

**Comment by BaseMax** [comment]
If you look at the Iranian Telegram app, you will see they have more than 50 different proxies on the list and after a few days, most proxies will die. And It's hard for them to find good proxies in such that long list. And it's why I suggested this feature.

Best Wishes

**Comment by pouriaksrvi** [comment]
> I have not talked about using AWS

If your request isn't about Amazon services, how is it related to "removing Iran from USA internet sanctions" as you mentioned in the original post?

**Comment by pouriaksrvi** [comment]
Anyway you may wanna like these two, too:
https://github.com/telegramdesktop/tdesktop/issues/25970
https://github.com/telegramdesktop/tdesktop/issues/25971

**Comment by BaseMax** [comment]
> Based on removing Iran from USA internet sanctions

Yes, This issue is older

Best wishes.

**Comment by pouriaksrvi** [comment]
> This issue is older

No. Those are two issues not one. Also, the both of them has been sent in the same time. 
By "This issue is older" to which issue did you refer?

**Comment by Anton-V-K** [comment]
I wonder why this feature still isn't available on desktop. It is a pain in the neck to watch for connection loss and switch proxy manually.
BTW auto-switching proxy works fine on Android.
See also https://github.com/telegramdesktop/tdesktop/issues/24713

**Comment by vgdh** [comment]
That would be great!

**Comment by saimyatthuztk2025-droid** [comment]
စိတ် ညှို့ရှင်

**Comment by SagePtr** [comment]
This looks implemented in 6.7.8 version, but for some reason in autoselects the proxy which is in bottom of the list from working proxies rather than the proxy with lowest ping.
